internal/handler/msg_saslstart.go

```diff
@@ -61,25 +65,41 @@ func (h *Handler) MsgSASLStart(ctx context.Context, msg *wire.OpMsg) (*wire.OpMs
 			return nil, err
 		}
 
+		conninfo.Get(ctx).SetAuth(username, password)
+
+		var emptyPayload types.Binary
+		must.NoError(reply.SetSections(wire.MakeOpMsgSection(
+			must.NotFail(types.NewDocument(
+				"conversationId", int32(1),
+				"done", true,
+				"payload", emptyPayload,
+				"ok", float64(1),
+			)),
+		)))
+
+	case "SCRAM-SHA-256":
+		response, err := h.saslStartSCRAMSHA256(ctx, document)
+		if err != nil {
+			return nil, err
+		}
+
+		conninfo.Get(ctx).BypassBackendAuth()
+
+		must.NoError(reply.SetSections(wire.MakeOpMsgSection(
+			must.NotFail(types.NewDocument(
+				"ok", float64(1),
+				"conversationId", int32(1),
+				"done", false,
+				"payload", response,
```

Comment:
I checkout the PR and was unable to authenticate because the payloads in the responses need to be binary. The tweak below allows a client to successfully authenticate with the exception of the legacy shell due to an invalid salt length and the dotnet driver because it uses the `skipEmptyExchange` option to avoid an extra round trip.

```go
	b := types.Binary{}
	b.B = []byte(resp)
	b.Subtype = 0

	var reply wire.OpMsg
	must.NoError(reply.SetSections(wire.MakeOpMsgSection(
		must.NotFail(types.NewDocument(
			"conversationId", int32(1),
			"done", true,
			"payload", b,
			"ok", float64(1),
		)),
	)))
```

With the exception of the legacy shell due to an invalid salt length and the dotnet driv

Reply:
Just ignore what `skipEmptyExchange` means for now as it requires further investigation.

Reply:
Does this mean existing integration tests are not sufficient? :thinking: I mean manual testing is great, but I thought such scenario is covered by integration tests

Reply:
Yeah that's a nice to have.

Reply:
Sorry nit: the payload variable is called `response` here, but `resp` in msg_saslcontinue.go.